internal/kafka/internal/workers/kafka_mgrs/promotion/internal/actions/update_kafka_request_action.go

```diff
@@ -0,0 +1,37 @@
+package actions
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/workers/kafka_mgrs/promotion/chain"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+)
+
+var _ chain.ReconcileAction[PromotionContext] = &UpdateKafkaRequestAction{}
+
+type UpdateKafkaRequestAction struct {
+	kafkaService services.KafkaService
+}
+
+func NewUpdateKafkaRequestAction(kafkaService services.KafkaService) chain.ReconcileAction[PromotionContext] {
+	return &UpdateKafkaRequestAction{
+		kafkaService: kafkaService,
+	}
+}
+
+func (u UpdateKafkaRequestAction) PerformJob(kafkaRequest *dbapi.KafkaRequest, currentResult chain.Result[PromotionContext]) (chain.Result[PromotionContext], bool, error) {
```

Comment:
We should document what is this action intended to perform along with the other actions.
Important in this cases to document what are the expected previous results, the assumptions made (like in this one it is that the kafka has successfully been reserved at ams level without error?), etc...